sat/cli/bootsys/bos.py

```diff
@@ -255,10 +255,16 @@ def has_completed(self):
                 # Only log progress update when components succeed or fail
                 self.pct_successful = float(self.session_status['percent_successful'])
                 self.pct_failed = float(self.session_status['percent_failed'])
-                LOGGER.info(
-                    'Session %s: %.0f%% components succeeded, %.0f%% components failed',
-                    self.bos_session_thread.session_id, self.pct_successful, self.pct_failed
-                )
+                if self.pct_successful > 99.0 and self.pct_successful < 100.0:
+                    LOGGER.info(
+                        'Session %s: %.10f%% components succeeded, %.2f%% components failed',
```

Comment:
10 decimal places is probably overkill. According to [the docs](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting), the default value of precision is 6. In order for a `self.pct_successful` value which is actually below 100 to be rounded up to 100 when rounding to 6 decimal places, the value would have to be greater than or equal to 99.9999995. That would mean only 5 out of a billion components were not yet successful in a BOS session. I don't expect we will ever boot a billion components at once with BOS.

Reply:
Yaa that's right Ryan, I had just considered it for an extreme condition and had taken it as 10. so, should be fine to have it considered at 6 itself as maximum decimal place.